apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/UrlSigner.CredentialBlobSigner.cs

```diff
@@ -12,33 +12,34 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-using Google.Apis.Auth.OAuth2;
+using Google.Api.Gax;
 using System.Threading;
 using System.Threading.Tasks;
+using ISignerCredential = Google.Apis.Auth.OAuth2.IBlobSigner;
 
 namespace Google.Cloud.Storage.V1
 {
     public sealed partial class UrlSigner
     {
         /// <summary>
-        /// Implementation of <see cref="IBlobSigner"/> to sign a blob from a service account credential.
-        /// We already have the private key, so don't need to perform any IO.
+        /// Implementation of <see cref="IBlobSigner"/> to sign a blob from a supported credential.
         /// </summary>
-        private sealed class ServiceAccountCredentialBlobSigner : IBlobSigner
+        private sealed class CredentialBlobSigner : IBlobSigner
         {
-            private readonly ServiceAccountCredential _credential;
+            private readonly ISignerCredential _credential;
 
-            internal ServiceAccountCredentialBlobSigner(ServiceAccountCredential credential) =>
-                _credential = credential;
+            internal CredentialBlobSigner(ISignerCredential credential, string id, string algorithm) =>
```

Comment:
I think if this took a `Func<Task<string>>` as an asynchronous ID provider, we could avoid some factory methods being async.